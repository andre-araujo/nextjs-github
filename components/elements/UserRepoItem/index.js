import {
    Flex,
    Box,
} from 'grid-styled';
import {
    number,
    string,
} from 'prop-types';

import {
    Anchor,
    Wrapper,
    Title,
    Starts,
    Language,
    Description,
} from './styles';

const UserRepoItem = ({
    language,
    stars,
    title,
    description,
    githubUrl,
}) => (
    <Wrapper>
        <Box mb="1em">
            <Anchor target="_blank" href={githubUrl}>
                <Title>
                    {title}
                    <img
                        width="15"
                        alt="github logo"
                        src="/static/images/github-logo.svg"
                    />
                </Title>
            </Anchor>
        </Box>
        <Description>{description}</Description>
        <Flex>
            <Starts>â˜… {stars}</Starts>
            <Language>{language}</Language>
        </Flex>
    </Wrapper>
);

UserRepoItem.defaultProps = {
    title: null,
    stars: null,
    description: null,
    githubUrl: null,
};

UserRepoItem.propTypes = {
    title: string.isRequired,
    stars: number,
    description: string,
    githubUrl: string,
};

export default UserRepoItem;
