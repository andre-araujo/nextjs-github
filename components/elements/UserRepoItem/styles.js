import styled from 'styled-components';

export const Wrapper = styled.li`
    border-bottom: 1px solid #eee;
    padding: 1.5rem 0;
`;

export const Anchor = styled.a`
    cursor: pointer;
    text-decoration: none;
`;

export const Title = styled.h3`
    color: #333;
    display: inline-block;
    font-family: AvantGardeDemi;
    font-size: 1rem;

    :hover {
        opacity: .6;
    }

    img {
        margin-left: .8rem;
    }
`;

export const Starts = styled.p`
    font-family: AvantGardeMedium;
    font-size: .7rem;
    margin-right: .5rem;
`;

export const Language = styled.p`
    font-family: AvantGardeLight;
    font-size: .8rem;
`;

export const Description = styled(Language)`
    margin-bottom: 1rem;
`;
