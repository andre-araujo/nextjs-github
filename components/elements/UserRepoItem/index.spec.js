import React from 'react';
import { shallow } from 'enzyme';

import UserRepoItem from './index';

describe('UserRepoItem component', () => {
    let props;

    beforeEach(() => {
        props = {
            description: 'description',
            githubUrl: 'githubUrl',
            stars: 321,
            title: 'title',
            fullname: 'some/name',
        };
    });

    it('should match snapshot with Unavaliable description', () => {
        expect(shallow(
            <UserRepoItem
                {...props}
            />,
        )).toMatchSnapshot();
    });
});
