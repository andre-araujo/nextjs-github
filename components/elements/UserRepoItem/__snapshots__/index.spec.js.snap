// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepoItem component should match snapshot 1`] = `
<styled.li>
  <Box
    mb="1rem"
  >
    <Link
      as="/repos/some/name"
      href="/repos?username=some&reponame=name"
    >
      <styled.a>
        <styled.h3>
          title
        </styled.h3>
      </styled.a>
    </Link>
    <styled.a
      href="githubUrl"
      target="_blank"
    >
      <img
        alt="github logo"
        src="/static/images/github-logo.svg"
        width="15"
      />
    </styled.a>
  </Box>
  <Styled(styled.p)>
    description
  </Styled(styled.p)>
  <Flex>
    <styled.p>
      â˜… 
      321
    </styled.p>
    <styled.p />
  </Flex>
</styled.li>
`;
